src/test/server/ServerStatus_test.cpp

```diff
@@ -0,0 +1,100 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/test/jtx.h>
+#include <beast/core/to_string.hpp>
+#include <beast/http.hpp>
+#include <boost/asio.hpp>
+
+namespace ripple {
+namespace test {
+
+class ServerStatus_test : public beast::unit_test::suite
+{
+public:
+    void
+    run()
+    {
+        using namespace jtx;
+        Env env(*this, []()
+            {
+                auto p = std::make_unique<Config>();
+                setupConfigForUnitTests(*p);
+                p->section("port_ws").set("protocol", "ws2,wss2");
+                return p;
+            }());
+        auto const port = env.app().config()["port_ws"].
+            get<std::int32_t>("port");
+        BEAST_EXPECT(port);
+
+        using namespace boost::asio;
+        using namespace beast::http;
+        io_service ios;
+        ip::tcp::resolver r{ios};
+        request_v1<empty_body> req;
+        req.method = "GET";
+        req.url = "/";
+        req.version = 11;
+        req.headers.replace("Host", "127.0.0.1:" + to_string(*port));
+        req.headers.replace("User-Agent", "Beast");
```

Comment:
User-Agent should be "ripple" or "test" or something but not Beast. Not a showstopper.